compiler/AST/symbol.cpp

```diff
@@ -144,10 +144,15 @@ static inline void verifyInTree(BaseAST* ast, const char* msg) {
 }
 
 void Symbol::verify() {
-  if (defPoint && !defPoint->parentSymbol && !toModuleSymbol(this))
-    INT_FATAL(this, "Symbol::defPoint is not in AST");
-  if (defPoint && this != defPoint->sym)
-    INT_FATAL(this, "Symbol::defPoint != Sym::defPoint->sym");
+  if (defPoint) {
+    if (!defPoint->parentSymbol && this != rootModule)
+      INT_FATAL(this, "Symbol::defPoint is not in AST");
+    if (this != defPoint->sym)
+      INT_FATAL(this, "Symbol::defPoint != Sym::defPoint->sym");
+  } else {
+    if (this != rootModule)
+      INT_FATAL(this, "Symbol without a defPoint");
+  }
```

Comment:
I think this section might be clearer if you separated out the this != rootModule check.

```
if ( this == rootModule ) {
  if (defPoint != NULL && this != defPoint->sym)
    INT_FATAL(this, "Symbol::defPoint != Sym::defPoint->sym");
} else {
  if (defPoint == NULL)
    INT_FATAL(this, "Symbol without a defPoint");
  if (defPoint->parentSymbol == NULL)
    INT_FATAL(this, "Symbol::defPoint is not in AST");
  if (this != defPoint->sym)
    INT_FATAL(this, "Symbol::defPoint != Sym::defPoint->sym"); 
}
```

But even then I am a bit confused - does rootModule have DefPoint or not? (I would expect it not to have a defPoint). In that case, you could simply skip all of this error checking if this == rootModule.